tango/step.py

```diff
@@ -432,6 +432,14 @@ def from_params(  # type: ignore[override]
 
         return subclass(step_name=step_name, step_config=raw_step_config, **kwargs)
 
+    # We only want one step object per step with a given unique id.
+    def __copy__(self):
+        return self
+
+    # We only want one step object per step with a given unique id.
+    def __deepcopy__(self, memo=None):
```

Comment:
Can we put a log warning that this is being overridden? I don't like that the expected behavior is being changed quietly.